core/src/main/java/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -243,7 +247,6 @@ private boolean copyBytes(File file, InputStream in, RandomAccessFile out) throw
                 offset = writeBlock(bytes, out, block, offset);
                 if (!verifyFormat.isVerifiedFormat() && (offset >= 1048576 || offset >= remoteSize)) { //let's check format after we get 1MB or full file
                     verifyFormat.invoke();
-                    if (verifyFormat.isInvalid()) return true;
```

Comment:
@Spaceman1984 cc @nvazquez why are we not validating format now?

Reply:
@rhtyd If validation fails, an exception is thrown, so there is no setting of an invalid state that has to be checked.

Reply:
In short, prior to these changes the invoke() method only set the boolean variable retrieved in case of valid/invalid. After these changes, if it is invalid invoke() throws an exception, so no need to check isInvalid() in this line